contrib/uservices/badge/src/configuration.rs

```diff
@@ -0,0 +1,125 @@
+use config::{Config, ConfigError, Environment, File};
+use sdk_cds::service::APIConfiguration;
+
+#[derive(Default, Debug, Deserialize, Clone)]
+#[serde(default)]
+pub struct Configuration {
+    pub badge: BadgeConfiguration,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone)]
+#[serde(default)]
+pub struct BadgeConfiguration {
+    pub url: String,
+    pub name: String,
+    pub mode: String,
+    pub api: APIConfiguration,
+    pub database: DatabaseConfiguration,
+    pub kafka: KafkaConfiguration,
+    pub http: HTTPConfiguration,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone)]
+#[serde(default)]
+pub struct DatabaseConfiguration {
+    pub user: String,
+    pub password: String,
+    pub name: String,
+    pub host: String,
+    pub port: i32,
+    pub sslmode: String,
+    pub maxconn: i32,
+    pub timeout: i32,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone)]
+#[serde(default)]
+pub struct KafkaConfiguration {
+    pub group: String,
+    pub user: String,
+    pub password: String,
+    pub broker: String,
+    pub topic: String,
+}
+
+#[derive(Default, Debug, Deserialize, Serialize, Clone)]
+pub struct HTTPConfiguration {
```

Comment:
Can be replaced by [std::net::SocketAddr](https://doc.rust-lang.org/std/net/enum.SocketAddr.html) which implement Serialize & Deserialize

Reply:
http_server(state: WebState, http_bind: String, http_port: String) ======> http_server(state: WebState, addr: SocketAddr)
with format!("{}:{}", http_bind, http_port).into()